api/src/neo4j/queries/relatedReactions.js

```diff
@@ -87,8 +88,9 @@ CALL apoc.cypher.run("
  
   MATCH (:Reaction${m} {id: $rid})-[cmE${v}]-(cm:CompartmentalizedMetabolite)
   WITH DISTINCT cm, cmE
-  MATCH (cm)-[${v}]-(:Metabolite)-[${v}]-(ms:MetaboliteState)
   MATCH (cm)-[${v}]-(c:Compartment)-[${v}]-(cs:CompartmentState)
+  USING JOIN on c
+  OPTIONAL MATCH (cm)-[${v}]-(:Metabolite)-[${v}]-(ms:MetaboliteState)
```

Comment:
I wasn't expected to see the `OPTIONAL` here, is there special reason for it?

Reply:
Without it the request takes about twice as long.

Reply:
That's really weird, it must mean that it chooses to traverse the network differently.
How would the results be affected if there is no match for that `optional` clause?

Reply:
I would guess that it returns the last part of the query as `{id: $rid, metabolites: []}` since we are performing a list aggregate operation, so it should be safe. In theory there shouldn't be any `CompartmentalizedMetabolite` that is not connected to any `Metabolite`.

I actually ran a `PROFILE` to compare the query with and without `OPTIONAL` now and the output plans are identical. I even saved them as PNGs and used a tool to compare the images to make sure.

We have previously used similar patterns with `OPTIONAL` in certain queries and they made a positive impact, which is why I tried it here as well.

Reply:
Thanks for sharing these details, `OPTIONAL` might come in handy. Personally, I'm still skeptical about (over)using it, but in this case it seems to do a good job.